tests/tracing/test_types.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+import mlflow
+from mlflow.tracing.types.model import SpanType
+
+
+def test_json_deserialization(mock_client):
+    class TestModel:
+        @mlflow.trace()
+        def predict(self, x, y):
+            z = x + y
+            z = self.add_one(z)
+            return z  # noqa: RET504
+
+        @mlflow.trace(
+            span_type=SpanType.LLM, name="add_one_with_custom_name", attributes={"delta": 1}
```

Comment:
Could we add some un-json-serializable thing in attributes so we can test that it doesn't throw as well?

```suggestion
            span_type=SpanType.LLM, name="add_one_with_custom_name", attributes={"delta": 1, "time": datetime.now()}
```

Reply:
Ooh actually turns out attributes can only be one of these (basically serializable) in OTel....
```
Invalid type datetime for attribute 'time' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types
```
We can support those unserializable data by some hack, but I think it's ok at this moment. We can add support anytime when it is requested from preview:)